src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -1241,18 +1171,238 @@ class Peer(
+                            swapInCommands.trySend(SwapInCommand.UnlockWalletInputs(cmd.walletInputs.map { it.outPoint }.toSet()))
+                        }
+                    }
+                }
+            }
+            is OpenOrSplicePayment -> {
+                val channel = channels.values.firstOrNull { it is Normal }
+                val currentFeerates = peerFeeratesFlow.filterNotNull().first()
+                val currentFeeCredit = feeCreditFlow.first().truncateToSatoshi()
+                when {
+                    channelFundingIsInProgress() -> {
+                        // Once the channel funding is complete, we may have enough inbound liquidity to receive the payment without an on-chain operation
+                        // This would be more economical, so we simply ignore the funding request.
+                        logger.warning { "ignoring on-the-fly funding, funding is already in progress" }
+                    }
+                    channel is Normal -> {
+                        // We don't contribute any input or output, but we must pay on-chain fees for the shared input and output.
+                        // We pay those on-chain fees using our current channel balance.
+                        val localBalance = channel.commitments.active.first().localCommit.spec.toLocal
+                        val spliceWeight = FundingContributions.computeWeightPaid(isInitiator = true, commitment = channel.commitments.active.first(), walletInputs = listOf(), localOutputs = listOf())
+                        val (fundingFeerate, localMiningFee) = client.computeSpliceCpfpFeerate(channel.commitments, currentFeerates.fundingFeerate, spliceWeight, logger)
+                        val (targetFeerate, paymentDetails) = when {
+                            localBalance >= localMiningFee + cmd.leaseFees(fundingFeerate).total -> {
+                                // We have enough funds to pay the mining fee and the lease fees.
+                                // This the ideal scenario because the fees can be paid immediately with the splice transaction.
+                                Pair(fundingFeerate, LiquidityAds.PaymentDetails.FromChannelBalance)
+                            }
+                            else -> {
+                                val targetFeerate = when {
+                                    localBalance >= localMiningFee * 0.75 -> fundingFeerate
+                                    // Our current balance is too low to pay the on-chain fees.
+                                    // We will consume all of it in on-chain fees, and target a higher feerate.
+                                    // This ensures that the resulting feerate won't be too low compared to our target.
+                                    // We must cover the shared input and the shared output, which is a lot of weight, so we add 50%.
+                                    else -> fundingFeerate * 1.5
+                                }
```

Comment:
I don't get this

Reply:
The wallet must pays on-chain fees during the `interactive-tx` session for its weight of the transaction (it must pay `localMiningFee`). Since we're not adding any additional on-chain inputs, the only way to pay those fees is from our channel balance. If our balance is too low, we will pay less fees than expected, so the resulting feerate of the transaction will be lower than our target. Since we cannot do anything about it (we don't have funds to pay for it), we instead tell the LSP to target a **higher** feerate:

- we won't be able to contribute enough, but we'll contribute everything we have
- the LSP will contribute more fees (because we asked for a higher feerate)
- so the resulting feerate shouldn't be too low

Does that make more sense?

Reply:
I've reworked the comment in the latest rebase, I hope it's clearer.